src/System.Diagnostics.EventLog/src/System/Diagnostics/EventLog.cs

```diff
@@ -899,14 +899,14 @@ internal static string UnsafeTryFormatMessage(SafeLibraryHandle hModule, uint me
                         messageNum,
                         0,
                         buf,
-                        buf.Capacity,
+                        buf.Length,
                         addresses);
 
                     if (msgLen == 0)
                     {
                         lastError = Marshal.GetLastWin32Error();
                         if (lastError == Interop.Kernel32.ERROR_INSUFFICIENT_BUFFER)
-                            buf.Capacity = buf.Capacity * 2;
+                            buf = new char[buf.Length * 2];
```

Comment:
Could this overflow (i.e. integer overflow)? Do we care?

Reply:
Yes, in theory. And if it did, you would get `System.OverflowException` that is about as good as any other exception here.